framework/test/gtest/ServiceReferenceTest.cpp

```diff
@@ -0,0 +1,115 @@
+/*=============================================================================
+
+Library: CppMicroServices
+
+Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+file at the top-level directory of this distribution and at
+https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=============================================================================*/
+
+#include "cppmicroservices/FrameworkFactory.h"
+#include "cppmicroservices/Framework.h"
+#include "cppmicroservices/FrameworkEvent.h"
+#include "cppmicroservices/BundleContext.h"
+#include "cppmicroservices/ServiceObjects.h"
+#include "gtest/gtest.h"
+
+using namespace cppmicroservices;
+
+namespace ServiceNS
+{
+
+  struct ITestServiceA
+  {
+    virtual int getValue() const = 0;
+    virtual ~ITestServiceA() {}
+  };
+
+  struct ITestServiceB
+  {
+    virtual int getValue() const = 0;
+    virtual ~ITestServiceB() {}
+  };
+
+
+}
+
+// This test exercises the 2 ways to register a service 
+//   a. using the name of the interface i.e. "Foo::Bar"
+//   b. using the type of the interface i.e. <Foo::Bar>
+// multiplied with 2 ways to get the service reference
+//   a. using the name of the interface
+//   b. using the type of the interface
+// NOTE: We do have tests for this in different places,
+// but this test consolidates all these modes together.
```

Comment:
Since we already have tests in different places can the other, redundant, tests be removed in favor of these tests?

Reply:
Jeff, what I meant was that the fragments exercising the same functionality do exist, but they are usually parts of a larger test.

Whereas this test just exercises the 2 variants of RegisterService x 2 variants of GetService (to ensure the functionality isn't affected). I agree that the test I added is redundant - i don't mind removing it, I guess.

Reply:
I'm not suggesting to remove the tests you've written. I'm asking if any of the tests which already exist can be removed in favor of this test.

Reply:
I've gone through them and it seems they are integral to the existing tests i.e. they can't be removed.